### DISCUSSION TITLE
Is it a good choice to parse a grammar?

### DISCUSSION THREAD

**Discussion by aidarnouman** [pointfreeco/swift-parsing#269] [General]
Hey guys! First of all, thanks for an amazing library.

I am exploring ways to do lexing and then parsing into AST. 
Valid syntax is something like numi.app / numbr.dev, i.e. intuitive notepad calculator kind of thing.

Looks like I *can* do it with swift-parsing, but maybe there are some performance quirks / pitfalls I am overlooking here. Alternative is to do it all *by hand*, just the usual scanning/parsing thing.

Appreciate any advice / opinion

**Comment by mbrandonw** [comment]
Hi @nugmanoff, it definitely is possible, and we have a few examples of this in our benchmark suite:

* Arithmetic: https://github.com/pointfreeco/swift-parsing/blob/main/Sources/swift-parsing-benchmark/Arithmetic.swift
* JSON: https://github.com/pointfreeco/swift-parsing/blob/main/Sources/swift-parsing-benchmark/JSON.swift

You could play around with those to see if they do what you want and have the performance characteristics you need.

**Reply by aidarnouman** [reply]
Thank you! I will give it a shot then!
